jsonnet/kube-prometheus/kube-prometheus.libsonnet

```diff
@@ -17,6 +18,11 @@ local kubeRbacProxyContainer = import './kube-rbac-proxy/container.libsonnet';
 (import './alerts/alerts.libsonnet') +
 (import './rules/rules.libsonnet') +
 {
+  nodeExporter: nodeExporter({
+    namespace: $._config.namespace,
+    version: '1.0.1',
+    imageRepo: 'quay.io/prometheus/node-exporter',
```

Comment:
if we're already refactoring, I would suggest to just have this be plain `image`, we tried this kind of "abstraction" in many ways, but we've learned that the image should just be treated as an opaque string

Reply:
I thought about it and applied such approach in new kube-rbac-proxy function. However in case of node-exporter I left it as is due to `app.kubernetes.io/version` label. I can see 3 options of satisfying label and image string:
1) current solution - compose `image` inside a function and pass `version` directly to label.
2) pass `image` string to a function and extract `version` from that string
3) pass `image` and `version` as strings.
I think 3rd one is the most fool-proof, but at the same time not so clean. That said I don't have any preference.

Reply:
I prefer 3, along the lines of, "prefer duplication over the wrong abstraction".

Reply:
Tested the 3rd approach in https://github.com/prometheus-operator/kube-prometheus/pull/862/commits/47f81f46f71a8cb672978bd17242b673472d8193#diff-46fb519eb228dee2044be788ed466089f6652ca4ea439f6f401dd3b509d12dcaR29-R30 and it looks a bit better and cleaner so I will change it here too.